systemtest/src/test/java/io/strimzi/systemtest/MultipleNamespaceST.java

```diff
@@ -0,0 +1,180 @@
+        );
+
+        checkRecordsForConsumer(MESSAGE_COUNT, jobReadMessagesForTarget);
+        deleteNewTopic();
+    }
+
+    /**
+     * Test the case when Kafka will be deployed in different namespace
+     */
+    @Test
+    @Tag(REGRESSION)
+    void testDeployKafkaToOtherNamespace() throws InterruptedException {
+        String kafkaName = kafkaClusterName(CLUSTER_NAME + "-second");
+
+        secondNamespaceResources.kafkaEphemeral(CLUSTER_NAME + "-second", 3).done();
+
+        LOGGER.info("Waiting for creation {} in namespace {}", kafkaName, SECOND_NAMESPACE);
+        kubeClient.namespace(SECOND_NAMESPACE);
+        kubeClient.waitForStatefulSet(kafkaName, 3);
+    }
+
+
+    /**
+     * Test the case when MirrorMaker will be deployed in different namespace across multiple namespaces
+     */
+    @Test
+    @Tag(REGRESSION)
+    void testDeployMirrorMakerAcrossMultipleNamespace() throws InterruptedException {
+        String kafkaName = CLUSTER_NAME + "-target";
+        String kafkaSourceName = kafkaClusterName(CLUSTER_NAME);
+        String kafkaTargetName = kafkaClusterName(kafkaName);
+
+        secondNamespaceResources.kafkaEphemeral(kafkaName, 3).done();
+        secondNamespaceResources.kafkaMirrorMaker(CLUSTER_NAME, kafkaSourceName, kafkaTargetName, "my-group", 1, false).done();
+
+        LOGGER.info("Waiting for creation {} in namespace {}", CLUSTER_NAME + "-mirror-maker", SECOND_NAMESPACE);
+        kubeClient.namespace(SECOND_NAMESPACE);
+        kubeClient.waitForDeployment(CLUSTER_NAME + "-mirror-maker", 1);
+    }
+
+
+
+    @BeforeAll
+    static void createClassResources(TestInfo testInfo) {
+        LOGGER.info("Creating resources before the test class");
+        classResources = new Resources(namespacedClient());
+        classResources().kafkaEphemeral(CLUSTER_NAME, 3)
+            .editSpec()
+                .editEntityOperator()
+                    .editTopicOperator()
+                        .withWatchedNamespace(SECOND_NAMESPACE)
+                    .endTopicOperator()
+                .endEntityOperator()
+            .endSpec()
+            .done();
+
+        testClass = testInfo.getTestClass().get().getSimpleName();
+    }
+
+    @AfterAll
+    static void deleteClassResources() {
+        LOGGER.info("Deleting resources after the test class");
+        classResources.deleteResources();
+        classResources = null;
+    }
+
+    @BeforeEach
+    void createSecondNamespaceResources() {
+        kubeClient.namespace(SECOND_NAMESPACE);
+        secondNamespaceResources = new Resources(namespacedClient());
+        kubeClient.namespace(DEFAULT_NAMESPACE);
+    }
+
+    @AfterEach
+    void deleteSecondNamespaceResources() {
+        secondNamespaceResources.deleteResources();
+        kubeClient.namespace(DEFAULT_NAMESPACE);
+    }
+
+    private static Resources classResources() {
+        return classResources;
+    }
+
+    private void deployNewTopic() {
```

Comment:
The tests would be eaiser to read if this was explicit about which namespace the topic was created in.

Reply:
Namespace and topic name added as an argument.